code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -1,339 +1,298 @@
-				return
 
-			user.visible_message("<span class='notice'>[user] cuts through the bolt covers on [src].</span>",
-			"<span class='notice'>You cut through the bolt cover.</span>")
-			removal_stage = 1
+/obj/machinery/nuclearbomb/stop_processing()
+	countdown.stop()
+	return ..()
 
-		if(1)
-			if(!iscrowbar(I))
-				return
 
-			user.visible_message("<span class='notice'>[user] starts forcing open the bolt covers on [src].</span>",
-			"<span class='notice'>You start forcing open the anchoring bolt covers with [I].</span>")
-			
-			if(!do_after(user, 15, TRUE, src, BUSY_ICON_BUILD))
-				return
+/obj/machinery/nuclearbomb/proc/explode()
+	stop_processing()
 
-			user.visible_message("<span class='notice'>[user] forces open the bolt covers on [src].</span>",
-			"<span class='notice'>You force open the bolt covers.</span>")
-			removal_stage = 2
+	if(safety)
+		timer_enabled = FALSE
+		GLOB.nukes_set_list -= src
+		return
 
-		if(2)
-			if(!iswelder(I))
-				return
+	if(exploded)
+		return
+	exploded = TRUE
+	safety = TRUE
+	if(!lighthack)
+		icon_state = "nuclearbomb3"
+
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_NUKE_EXPLODED)
+	addtimer(CALLBACK(src, .proc/BurnEveryoneOnZLevel, z), 5 SECONDS)
+	play_cinematic() //The round ends as soon as this happens, or it should.
+	return TRUE
 
-			var/obj/item/tool/weldingtool/WT = I
 
-			user.visible_message("<span class='notice'>[user] starts cutting apart the anchoring system sealant on [src].</span>",
-			"<span class='notice'>You start cutting apart the anchoring system's sealant with [I].</span>")
+/obj/machinery/nuclearbomb/proc/play_cinematic()
+	GLOB.enter_allowed = FALSE
+	priority_announce("DANGER. DANGER. Planetary Nuke Activated. DANGER. DANGER. Self destruct in progress. DANGER. DANGER.", "Priority Alert")
+	SEND_SOUND(world, pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg'))
 
-			if(!do_after(user, 40, TRUE, src, BUSY_ICON_BUILD, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)) || !WT.remove_fuel(5, user))
-				return
+	for(var/x in GLOB.player_list)
+		var/mob/M = x
+		if(isobserver(M))
+			continue
+		shake_camera(M, 110, 4)
 
-			user.visible_message("<span class='notice'>[user] cuts apart the anchoring system sealant on [src].</span>",
-			"<span class='notice'>You cut apart the anchoring system's sealant.</span>")
-			removal_stage = 3
+	Cinematic(CINEMATIC_SELFDESTRUCT, world)
 
-		if(3)
-			if(!iswrench(I))
-				return
-			user.visible_message("<span class='notice'>[user] begins unwrenching the anchoring bolts on [src].</span>",
-			"<span class='notice'>You begin unwrenching the anchoring bolts.</span>")
 
-			if(!do_after(user, 50, TRUE, src, BUSY_ICON_BUILD))
-				return
+/obj/machinery/nuclearbomb/proc/BurnEveryoneOnZLevel(z)
+	if(!z)
+		return
+	for(var/mob/living/victim in GLOB.alive_mob_list)
```

Comment:
Ghosts and new players cannot be alive, this can be optimized.

Reply:
```
/mob/Initialize()
	GLOB.mob_list += src
	if(stat == DEAD)
		GLOB.dead_mob_list += src
	else
		GLOB.alive_mob_list += src
```
This line made me cautious.

Reply:
It should be safe to implement it then